File: backend/Domain/Models/WaiterCall.cs
using System.ComponentModel.DataAnnotations.Schema;
using System.Text.Json.Serialization;
using Domain.DataBaseAccess;
using Microsoft.EntityFrameworkCore;

namespace Domain.Models;

public class WaiterCall: ITableDataBase, ITakeRelatedData
{
    public long? Id { get; set; }
    [JsonPropertyName("client_id")] public long? ClientId { get; init; }
    [JsonPropertyName("table_id")] public long TableId { get; init; }
    [JsonPropertyName("date_time")] public DateTime DateTime { get; init; }
    public WaiterCallStatus Status { get; init; }
    [NotMapped] public Client? Client { get; set; }

    public enum WaiterCallStatus
    {
        InAssembly,
        Done
    }

    public async Task TakeRelatedData(ApplicationDbContext context)
    {
        Client = await context.Clients.FirstOrDefaultAsync(c => c.Id == ClientId);
    }
}

File: API/tg_bots/Model/WaiterCall.py
from pydantic import BaseModel
from typing import Optional
from datetime import datetime
from enum import Enum
from Model.Client import Client


class WaiterCallStatus(Enum):
    IN_ASSEMBLY = 0
    DONE = 1


class WaiterCall(BaseModel):
    id: Optional[int]
    client_id: int
    table_id: int
    date_time: datetime
    status: WaiterCallStatus
    client: Client

    class Config:
        use_enum_values = True

    def __init__(self,
                 client_id: int,
                 table_id: int,
                 date_time: datetime,
                 status: WaiterCallStatus,
                 client: Client,
                 id: Optional[int] = None):
        super().__init__(
            id=id,
            client_id=client_id,
            table_id=table_id,
            date_time=date_time,
            status=status,
            client=client
        )
